"stopBranch called more than once by remote on" exception crashing Bazel

### Description of the bug:


We are somewhat frequently seeing error messages like :
```00:01:03.956   126 FATAL: bazel crashed due to an internal error. Printing stack trace:
00:01:03.956   127 java.lang.AssertionError: stopBranch called more than once by remote on external/rules_jvm_external+/private/tools/java/com/github/bazelbuild/rules_jvm_external/zip/libzip.jar
00:01:03.956   128 	at com.google.devtools.build.lib.dynamic.DynamicSpawnStrategy.stopBranch(DynamicSpawnStrategy.java:623)
00:01:03.956   129 	at com.google.devtools.build.lib.dynamic.RemoteBranch.lambda$callImpl$1(RemoteBranch.java:160)
00:01:03.956   130 	at com.google.devtools.build.lib.exec.AbstractSpawnStrategy$SpawnExecutionContextImpl.lockOutputFiles(AbstractSpawnStrategy.java:363)
00:01:03.956   131 	at com.google.devtools.build.lib.exec.local.LocalSpawnRunner.exec(LocalSpawnRunner.java:164)
00:01:03.956   132 	at com.google.devtools.build.lib.remote.RemoteSpawnRunner.execLocally(RemoteSpawnRunner.java:563)
00:01:03.956   133 	at com.google.devtools.build.lib.remote.RemoteSpawnRunner.execLocallyAndUpload(RemoteSpawnRunner.java:710)
00:01:03.956   134 	at com.google.devtools.build.lib.remote.RemoteSpawnRunner.execLocallyAndUploadOrFail(RemoteSpawnRunner.java:598)
00:01:03.956   135 	at com.google.devtools.build.lib.remote.RemoteSpawnRunner.exec(RemoteSpawnRunner.java:249)
00:01:03.956   136 	at com.google.devtools.build.lib.exec.AbstractSpawnStrategy.exec(AbstractSpawnStrategy.java:161)
00:01:03.956   137 	at com.google.devtools.build.lib.dynamic.RemoteBranch.runRemotely(RemoteBranch.java:89)
00:01:03.956   138 	at com.google.devtools.build.lib.dynamic.RemoteBranch.callImpl(RemoteBranch.java:153)
00:01:03.956   139 	at com.google.devtools.build.lib.dynamic.Branch.call(Branch.java:188)
00:01:03.956   140 	at com.google.devtools.build.lib.dynamic.Branch.call(Branch.java:43)
00:01:03.956   141 	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:128)
00:01:03.956   142 	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:74)
00:01:03.956   143 	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:80)
00:01:03.956   144 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
00:01:03.956   145 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
00:01:03.956   146 	at java.base/java.lang.Thread.run(Unknown Source)```

We've seen this error with the same stack trace, other than a differing file ~50 times in the last month.
The files listed that stopBranch was called more than once by remote on are most frequently build tools, persistent worker executables, frequently used external jars, and compiler binaries. A few other example files it's hit on are `cake/closure_compiler.jar`, `external/rules_scala_annex+/third_party/bazel/src/java_tools/buildjar/java/com/google/devtools/build/buildjar/jarhelper/libjarhelper.jar`, and `cake/build/css/less_worker.js` if that's helpful at all.

We suspect that this might have something to do with dynamic spawn strategy, but are unsure. We are using buildfarm as our remote execution software.

### Which category does this issue belong to?

Remote Execution

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.


I spent some time trying to make a minimal example for reproduction, but was unable to reproduce outside of our CI. I'll keep working on this, and will update if I make any headway. 

### Which operating system are you running Bazel on?


Ubuntu 24.04.3

### What is the output of `bazel info release`?


release 8.4.0

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.


_No response_

### What's the output of `git remote get-url origin; git rev-parse HEAD` ?


```text

```

### If this is a regression, please try to identify the Bazel commit where the bug was introduced with bazelisk --bisect.


_No response_

### Have you found anything relevant by searching the web?

I've seen a few "stopBranch called more than once" errors online, (e.g. https://github.com/bazelbuild/bazel/issues/12364 ) but only with local, and that issue was closed as fixed. 
We've also started a discussion in the bazel thread workspace:
https://bazelbuild.slack.com/archives/CD4MDG09Z/p1761334548385729

### Any other information, logs, or outputs that you want to share?

_No response_

--- a/src/main/java/com/google/devtools/build/lib/dynamic/DynamicSpawnStrategy.java
+++ b/src/main/java/com/google/devtools/build/lib/dynamic/DynamicSpawnStrategy.java
@@ -672,13 +672,7 @@ public class DynamicSpawnStrategy implements SpawnStrategy {
     // value of "cancellingStrategy", we do not expect concurrent calls to this method. (If there
     // are, we are in big trouble.)
     DynamicMode current = strategyThatCancelled.get();
-    if (cancellingStrategy.equals(current)) {
-      throw new AssertionError(
-          "stopBranch called more than once by "
-              + cancellingStrategy
-              + " on "
-              + getSpawnReadableId(cancellingBranch.getSpawn()));
-    } else {
+    if (!cancellingStrategy.equals(current)) {
       // Protect against the two branches from cancelling each other. The first branch to set the
       // reference to its own identifier wins and is allowed to issue the cancellation; the other
       // branch just has to give up execution.

